Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift

```diff
@@ -314,43 +323,72 @@ extension SwiftPMBuildSystem: SKCore.BuildSystem {
 
   public var indexPrefixMappings: [PathPrefixMapping] { return [] }
 
-  public func buildSettings(for uri: DocumentURI, language: Language) throws -> FileBuildSettings? {
-    // SwiftPMBuildSystem doesn't respect the langue specified by the editor.
-    return try buildSettings(for: uri)
-  }
-
-  private func buildSettings(for uri: DocumentURI) throws -> FileBuildSettings? {
-    guard let url = uri.fileURL else {
+  public func buildSettings(
+    for uri: DocumentURI,
+    in configuredTarget: ConfiguredTarget,
+    language: Language
+  ) throws -> FileBuildSettings? {
+    guard let url = uri.fileURL, let path = try? AbsolutePath(validating: url.path) else {
       // We can't determine build settings for non-file URIs.
       return nil
     }
-    guard let path = try? AbsolutePath(validating: url.path) else {
-      return nil
-    }
 
-    if let buildTarget = try buildTarget(for: path) {
-      return FileBuildSettings(
-        compilerArguments: try buildTarget.compileArguments(for: path.asURL),
-        workingDirectory: workspacePath.pathString
-      )
+    if configuredTarget.targetID == "" {
+      return try settings(forPackageManifest: path)
     }
 
-    if path.basename == "Package.swift" {
-      return try settings(forPackageManifest: path)
+    let buildTargets = self.targets.filter({ $0.name == configuredTarget.targetID })
+    if buildTargets.count > 1 {
+      logger.error("Found multiple targets with name \(configuredTarget.targetID). Picking the first one")
+    }
+    guard let buildTarget = buildTargets.first else {
+      if buildTargets.isEmpty {
+        logger.error("Did not find target with name \(configuredTarget.targetID)")
+      }
+      return nil
     }
 
-    if path.extension == "h" {
-      return try settings(forHeader: path)
+    if url.pathExtension == "h", let substituteFile = buildTarget.sources.first {
```

Comment:
`hpp` is pretty common as well. Should this instead call `defaultLanguage` and then check if it ends with `-header`? (We'd have to implement something small for `defaultLanguage` here)

Reply:
I changed to just check if `url` is part of `sources` and if not, assume that it’s a header file. That removes the need for file extension checking, which should be more general.